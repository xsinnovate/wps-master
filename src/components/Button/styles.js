import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    height: 40,
    borderColor: '#333',
    borderRadius: 20,
    borderWidth: 1,
    paddingHorizontal: 15,
  },
  text: {
    color: '#333',
    fontSize: 16,
    fontWeight: '600',
    lineHeight: 40,
    textAlign: 'center',
  },
});
