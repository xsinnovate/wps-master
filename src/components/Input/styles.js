import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  label: {
    paddingBottom: 4,
  },
  labelText: {
    color: '#777',
  },
  input: {
    height: 40,
    borderColor: '#ccc',
    borderRadius: 5,
    borderWidth: 1,
    paddingHorizontal: 10,
  },
});
