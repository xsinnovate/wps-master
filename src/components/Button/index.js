import PropTypes from 'prop-types';
import React from 'react';
import { TouchableHighlight, Text, View } from 'react-native';
import styles from './styles';

export default function Button({ children, onPress }) {
  return (
    <TouchableHighlight
      activeOpacity={0.5}
      onPress={onPress}
      underlayColor="#fff"
    >
      <View style={styles.container}>
        <Text style={styles.text}>
          {children}
        </Text>
      </View>
    </TouchableHighlight>
  );
}

Button.propTypes = {
  children: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};
