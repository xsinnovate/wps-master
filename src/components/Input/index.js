import PropTypes from 'prop-types';
import React from 'react';
import { TextInput, Text, View } from 'react-native';
import styles from './styles';

export default function Input({
  label,
  onChange,
  value,
  ...rest
}) {
  return (
    <View>
      <View style={styles.label}>
        <Text style={styles.labelText}>
          {label}
        </Text>
      </View>
      <View>
        <TextInput
          {...rest}
          onChangeText={onChange}
          value={value}
          style={styles.input}
        />
      </View>
    </View>
  );
}

Input.propTypes = {
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};
