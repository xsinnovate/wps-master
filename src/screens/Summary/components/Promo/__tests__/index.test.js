import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../../../../components/Button';
import Input from '../../../../../components/Input';
import { Promo } from '..';

const noop = () => {};

describe('<Promo />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Promo applyPromoCode={noop} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls applyPromoCode function with value on submit', () => {
    const spy = jest.fn();
    const instance = renderer.create(<Promo applyPromoCode={spy} />).root;

    instance.findByType(Input).props.onChange('DISCOUNT');
    instance.findByType(Button).props.onPress();

    expect(spy).toBeCalledWith('DISCOUNT');
  });
});
