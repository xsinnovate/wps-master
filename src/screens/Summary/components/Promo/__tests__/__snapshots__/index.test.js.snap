// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Promo /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "paddingBottom": 4,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#777",
            }
          }
        >
          Promo code
        </Text>
      </View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="characters"
          autoCorrect={false}
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#ccc",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingLeft": 20,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "borderColor": "#333",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": 40,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 40,
              "textAlign": "center",
            }
          }
        >
          Apply
        </Text>
      </View>
    </View>
  </View>
</View>
`;
